src/util/cpuid.cc

```diff
@@ -0,0 +1,218 @@
+  cpu_vendor_id << (char)(cpuid[1] >> 24 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[3] & 0xFF);
+  cpu_vendor_id << (char)(cpuid[3] >> 8 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[3] >> 16 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[3] >> 24 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[2] & 0xFF);
+  cpu_vendor_id << (char)(cpuid[2] >> 8 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[2] >> 16 & 0xFF);
+  cpu_vendor_id << (char)(cpuid[2] >> 24 & 0xFF);
+  return cpu_vendor_id.str();
+}
+
+std::string CpuidCommand::GetCpuModelName()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(0x80000000, 0, cpuid))
+    {
+      return "";
+    }
+
+  CryptoPP::word32 max_supported = cpuid[0] & 0xFFFFFFFF;
+  if (max_supported < 0x80000004)
+    {
+      return "";
+    }
+
+  std::stringstream cpu_model_name;
+  for (CryptoPP::word32 i = 0x80000002; i <= 0x80000004; i++)
+    {
+      cpuid[4] = {0};
+
+      if (!CryptoPP::CpuId(i, 0, cpuid))
+        {
+          return "";
+        }
+
+      for (int j = 0; j < 4; j++)
+        {
+          cpu_model_name << (char)(cpuid[j] & 0xFF);
+          cpu_model_name << (char)(cpuid[j] >> 8 & 0xFF);
+          cpu_model_name << (char)(cpuid[j] >> 16 & 0xFF);
+          cpu_model_name << (char)(cpuid[j] >> 24 & 0xFF);
+        }
+    }
+  return cpu_model_name.str();
+}
+
+int CpuidCommand::GetCpuFamily()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(1, 0, cpuid))
+    {
+      return 0;
+    }
+
+  int cpu_family = (cpuid[0] >> 8) & 0xf;
+  return cpu_family;
+}
+
+int CpuidCommand::GetCpuExtendedFamily()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(1, 0, cpuid))
+    {
+      return 0;
+    }
+
+  int cpu_family = (cpuid[0] >> 20) & 0xff;
+  return cpu_family;
+}
+
+int CpuidCommand::GetCpuModel()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(1, 0, cpuid))
+    {
+      return 0;
+    }
+
+  int cpu_model = (cpuid[0] >> 4) & 0xf;
+  return cpu_model;
+}
+
+int CpuidCommand::GetCpuExtendedModel()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(1, 0, cpuid))
+    {
+      return 0;
+    }
+
+  int cpu_model = (cpuid[0] >> 16) & 0xf;
+  return cpu_model;
+}
+
+int CpuidCommand::GetCpuStepping()
+{
+  CryptoPP::word32 cpuid[4] = {0};
+  if (!CryptoPP::CpuId(1, 0, cpuid))
+    {
+      return 0;
+    }
+
+  return cpuid[0] & 0xf;
+}
+
+void CpuidCommand::PrintUsage(const std::string& name) const
+{
+  LOG(info) << "Syntax: " << name;
+}
+
+bool CpuidCommand::Impl(
+    const std::string& cmd_name,
+    const std::vector<std::string>& args)
+{
+#ifdef WITH_CRYPTOPP
+#if defined(__x86_64__) \
+    || defined(_M_X64)  // TODO(selsta): _M_IX86 and _X86_ ??
+  LOG(info) << "CPU Vendor: " << GetCpuVendor();
+  LOG(info) << "CPU Model Name: " << GetCpuModelName();
+  int cpu_family = GetCpuFamily();
+  int cpu_model = GetCpuModel();
+  if (cpu_family == 15
+      || (cpu_family == 6 && GetCpuVendor().compare("GenuineIntel") == 0))
```

Comment:
>15
>6

Constants should be `const` or `constexpr` variables with human-meaningful names instead of raw values.